packages/driver/test/cypress/integration/commands/actions/check_spec.js

```diff
@@ -0,0 +1,1233 @@
+        expect($inputs.filter(':checked').length).to.eq(2)
+
+        expect($inputs.filter('[value=blue],[value=green]')).to.be.checked
+      })
+    })
+
+    it('can forcibly click even when being covered by another element', () => {
+      const checkbox = $('<input type=\'checkbox\' />').attr('id', 'checkbox-covered-in-span').prependTo($('body'))
+
+      cy.$$('<span>span on checkbox</span>').css({ position: 'absolute', left: checkbox.offset().left, top: checkbox.offset().top, padding: 5, display: 'inline-block', backgroundColor: 'yellow' }).prependTo($('body'))
+
+      let clicked = false
+
+      checkbox.on('click', () => {
+        clicked = true
+      })
+
+      cy.get('#checkbox-covered-in-span').check({ force: true }).then(() => {
+        expect(clicked).to.be.true
+      })
+    })
+
+    it('passes timeout and interval down to click', (done) => {
+      const checkbox = $('<input type=\'checkbox\' />')
+      .attr('id', 'checkbox-covered-in-span')
+      .prependTo($('body'))
+
+      cy.$$('<span>span on checkbox</span>')
+      .css({
+        position: 'absolute',
+        left: checkbox.offset().left,
+        top: checkbox.offset().top,
+        padding: 5,
+        display: 'inline-block',
+        backgroundColor: 'yellow',
+      })
+      .prependTo($('body'))
+
+      cy.on('command:retry', _.once((options) => {
+        expect(options.timeout).to.eq(1000)
+        expect(options.interval).to.eq(60)
+
+        done()
+      }))
+
+      cy.get('#checkbox-covered-in-span').check({ timeout: 1000, interval: 60 })
+    })
+
+    it('waits until element is no longer disabled', () => {
+      const chk = $(':checkbox:first').prop('disabled', true)
+
+      let retried = false
+      let clicks = 0
+
+      chk.on('click', () => {
+        clicks += 1
+      })
+
+      cy.on('command:retry', _.after(3, () => {
+        chk.prop('disabled', false)
+        retried = true
+      }))
+
+      cy.get(':checkbox:first').check().then(() => {
+        expect(clicks).to.eq(1)
+
+        expect(retried).to.be.true
+      })
+    })
+
+    it('delays 50ms before resolving', () => {
+      cy.$$(':checkbox:first').on('change', (e) => {
+        cy.spy(Promise, 'delay')
+      })
+
+      cy.get(':checkbox:first').check().then(() => {
+        expect(Promise.delay).to.be.calledWith(50, 'click')
+      })
+    })
+
+    describe('assertion verification', () => {
+      beforeEach(() => {
+        cy.on('log:added', (attrs, log) => {
+          if (log.get('name') === 'assert') {
+            this.lastLog = log
+          }
+        })
+      })
+
+      it('eventually passes the assertion', () => {
+        cy.$$(':checkbox:first').click(function () {
+          _.delay(() => {
+            cy.$$(this).addClass('checked')
+          }
+          , 100)
+        })
+
+        cy.get(':checkbox:first').check().should('have.class', 'checked').then(() => {
+          const {
+            lastLog,
+          } = this
```

Comment:
`this` with arrow functions, is this maintaining the correct `this`?

Reply:
Yes, `lastLog` has the value of `this.lastLog` and is not empty.